app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -200,24 +196,31 @@ class HomeFragment : Fragment() {
             )
         )
 
-        sessionControlView = SessionControlView(homeFragmentStore, view.homeLayout, sessionControlInteractor)
-
-        ConstraintSet().apply {
-            clone(view.homeLayout)
-            connect(sessionControlView.view.id, TOP, view.wordmark.id, BOTTOM)
-            connect(sessionControlView.view.id, START, PARENT_ID, START)
-            connect(sessionControlView.view.id, END, PARENT_ID, END)
-            connect(sessionControlView.view.id, BOTTOM, view.bottom_bar.id, TOP)
-            applyTo(view.homeLayout)
-        }
+        view.homeAppBar.addOnOffsetChangedListener(
```

Comment:
nit: can we put these calls into helper methods? For extremely high-level methods like `onCreateView`, I actually prefer each call to happen at a high level – i.e. no imperative code, just calling into one line high-level functions like, `attachNotificationObservers()` and `attachAndUpdateAppBarLayout()`.

That being said, that's more of an ideal but if something is more than 4 lines or so, it should be a helper method. A best practice is for functions to fit on a single screen and for methods that have so much going on, they grow quickly.